### ISSUE TITLE
cart.updated_at doesn't get updated with the current date timestamp

### ISSUE THREAD

**Issue by metaphyser** [medusajs/medusa#5729]
### Describe the bug

Whenever the cart gets updated, the cart.updated_at timestamp doesn't change and has the same timestamp as before (a few seconds after the creation).

### System information

Medusa version (including plugins):
```
    "@medusajs/admin": "^6.0.0",
    "@medusajs/cache-inmemory": "^1.8.7",
    "@medusajs/cache-redis": "^1.8.7",
    "@medusajs/event-bus-local": "^1.9.4",
    "@medusajs/event-bus-redis": "^1.8.7",
    "@medusajs/medusa": "1.18",
    "@medusajs/medusa-cli": "^1.3.18",
    "@medusajs/types": "^1.10.1",
    "babel-preset-medusa-package": "^1.1.13",
    "body-parser": "^1.19.0",
    "cors": "^2.8.5",
    "dotenv": "^16.1.4",
    "express": "^4.17.2",
    "medusa-file-s3": "^1.2.1",
    "medusa-fulfillment-manual": "^1.1.37",
    "medusa-interfaces": "^1.3.7",
    "medusa-payment-manual": "^1.0.24",
    "medusa-payment-paypal": "^6.0.2",
    "medusa-payment-stripe": "^6.0.2",
    "medusa-plugin-algolia": "^0.2.20",
    "medusa-plugin-sendgrid": "^1.3.11",
    "typeorm": "^0.3.16"
```

Node.js version: v20.0.0
Database: Postgres
Operating system: macOS 13.3.1 (a) 

### Steps to reproduce the behavior

1. update any cart with /store/carts/{id}
2. the response cart will have the updated_at with the same timestamp as before (a few seconds after the creation)

### Expected behavior

The cart.updated_at represents the timestamp of when it was last updated

### Additional context

I'm not sure if it's relevant, but postgres was showing this error, but not anymore.
```
2023-11-22 12:32:12.645 GMT [27750] ERROR:  invalid input syntax for type timestamp with time zone: "0NaN-NaN-NaNTNaN:NaN:NaN.NaN+NaN:NaN"
2023-11-22 12:32:12.645 GMT [27750] CONTEXT:  unnamed portal parameter $1 = '...'
2023-11-22 12:32:12.645 GMT [27750] STATEMENT:  SELECT "Cart"."id" AS "Cart_id", "Cart"."created_at" AS "Cart_created_at", "Cart"."updated_at" AS "Cart_updated_at", "Cart"."deleted_at" AS "Cart_deleted_at", "Cart"."email" AS "Cart_email", "Cart"."billing_address_id" AS "Cart_billing_address_id", "Cart"."shipping_address_id" AS "Cart_shipping_address_id", "Cart"."region_id" AS "Cart_region_id", "Cart"."customer_id" AS "Cart_customer_id", "Cart"."payment_id" AS "Cart_payment_id", "Cart"."type" AS "Cart_type", "Cart"."completed_at" AS "Cart_completed_at", "Cart"."payment_authorized_at" AS "Cart_payment_authorized_at", "Cart"."idempotency_key" AS "Cart_idempotency_key", "Cart"."context" AS "Cart_context", "Cart"."metadata" AS "Cart_metadata", "Cart"."sales_channel_id" AS "Cart_sales_channel_id" FROM "public"."cart" "Cart" WHERE ( ("Cart"."updated_at" > $1 AND "Cart"."updated_at" <= $2) ) AND ( "Cart"."deleted_at" IS NULL )
```

**Comment by kocendavid** [comment]
I have same problem.

**Comment by SalahAdDin** [comment]
@metaphyser Is it fixed?

**Comment by metaphyser** [comment]
@SalahAdDin it wasn't when I last tested 10 months ago. I've moved on in the meantime.

**Comment by SalahAdDin** [comment]
> @SalahAdDin it wasn't when I last tested 10 months ago. I've moved on in the meantime.

What are you using now?

**Comment by metaphyser** [comment]
I just checked, I used the created_at which was the next best thing even though it is not ideal.

**Comment by HASSAN0109** [comment]
@SalahAdDin it wasn't when I last tested 10 months ago. I've moved on in
the meantime.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.

@SalahAdDin it wasn't when I last tested 10 months ago. I've moved on in
the meantime.

What are you using now?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.
I just checked, I used the created_at which was the next best thing even
though it is not ideal.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.
Thanks for your time today 🤠
